File: src/main/java/com/kayukin/reportviewer/configuration/ApplicationProperties.java
package com.kayukin.reportviewer.configuration;

import org.springframework.boot.context.properties.ConfigurationProperties;
import software.amazon.awssdk.regions.Region;

@ConfigurationProperties(prefix = "reporting")
public record ApplicationProperties(
        S3Configuration s3,
        String downloadLocation,
        String apiBaseUrl
) {
    public record S3Configuration(
            String bucket,
            Region region,
            S3AuthType authType,
            String profile,
            String accessKey,
            String secretKey
    ) {
    }

    public enum S3AuthType {
        STATIC,
        PROFILE,
        IRSA,
        DEFAULT
    }
}

File: src/main/kotlin/com/kayukin/reportviewer/configuration/ApplicationProperties.kt
package com.kayukin.reportviewer.configuration

import org.springframework.boot.context.properties.ConfigurationProperties
import software.amazon.awssdk.regions.Region

@ConfigurationProperties(prefix = "reporting")
data class ApplicationProperties(
    val s3: S3Configuration,
    val downloadLocation: String,
    val apiBaseUrl: String
) {
    data class S3Configuration(
        val endpoint: String?,
        val bucket: String,
        val region: Region,
        val authType: S3AuthType,
        val profile: String?,
        val accessKey: String?,
        val secretKey: String?
    )

    enum class S3AuthType {
        STATIC,
        PROFILE,
        IRSA,
        DEFAULT
    }
}